doc/build.rst

```diff
@@ -317,51 +290,33 @@ simplest way to install the R package after obtaining the source code is:
   cd R-package
   R CMD INSTALL .
 
-As an alternative, the package can also be loaded through ``devtools::load_all()``
-from the same subfolder ``R-package`` in the repository's root, and by extension, can
-be installed through RStudio's build panel if one adds that folder ``R-package`` as an
-R package project in the RStudio IDE.
+Use the environment variable ``MAKEFLAGS=-j$(nproc)`` if you want to speedup the build. As
+an alternative, the package can also be loaded through ``devtools::load_all()`` from the
+same subfolder ``R-package`` in the repository's root, and by extension, can be installed
+through RStudio's build panel if one adds that folder ``R-package`` as an R package
+project in the RStudio IDE.
 
-If you want to use the CMake build for better performance or greater flexibility around
-compile flags, the earlier snippet can be replaced by:
-
-.. code-block:: bash
-
-  cmake -B build -S . -DR_LIB=ON
-  cmake --build build --target install -j$(nproc)
-
-Note in this case that ``cmake`` will not take configurations from your regular ``Makevars``
-file (if you have such a file under ``~/.R/Makevars``) - instead, custom configurations such
-as compilers to use and flags need to be set through environment variables like ``${CC}``,
-``${CFLAGS}``, etc.
-
-Installing the development version with Visual Studio (Windows)
-===============================================================
-
-On Windows, CMake with Visual C++ Build Tools (or Visual Studio) can be used to build the R package.
-
-While not required, this build can be faster if you install the R package ``processx`` with ``install.packages("processx")``.
-
-.. note:: Setting correct PATH environment variable on Windows
+.. code-block:: R
 
-  If you are using Windows, make sure to include the right directories in the PATH environment variable.
+  library(devtools)
+  devtools::load_all(path = "/path/to/xgboost/R-package")
 
-  * If you are using R 4.x with RTools 4.0:
-    - ``C:\rtools40\usr\bin``
-    - ``C:\rtools40\mingw64\bin``
+If you want to use the CMake build for greater flexibility around compile flags, the
+earlier snippet can be replaced by:
 
-  * If you are using R 3.x with RTools 3.x:
+.. code-block:: bash
 
-    - ``C:\Rtools\bin``
-    - ``C:\Rtools\mingw_64\bin``
+  cmake -B build -S . -DR_LIB=ON -GNinja
```

Comment:
Wouldn't this make it use the MSVC compiler by default?

Reply:
There's a warning in the end of this section, MSVC with R is not supported.

Reply:
Perhaps it could be helpful to provide instructions for a clang install with a tool like conda to build the package with GPU support.

Reply:
Let me add a note that cmake is not supported either. I tried to make it work with Windows and the latest rtools, it's quite a hack. I don't expect users to do something similar.

The mingw is hidden deep in the rtools43 package.